_src-lp/blocks/b-productswithselectors/__TESTS__/b-productswithselectors.spec.js

```diff
@@ -0,0 +1,67 @@
+// eslint-disable-next-line import/extensions
+import mockData from './mock-data.json';
+// mock everything that is not related to the actual main scope
+jest.mock('../../../scripts/lib-franklin.js', () => ({
+  ...jest.requireActual('../../../scripts/lib-franklin.js'),
+  loadHeader: jest.fn(),
+  loadFooter: jest.fn(),
+}));
+
+describe('b-productswithselectors block', () => {
+
+  beforeEach(() => {
+    jest.resetModules();
+    jest.clearAllMocks();
+  });
+
+  describe('b-productswithselectors', () => {
+
+    // init / bootstrap page()
+    document.body.innerHTML = mockData.b_productswithselectors;
+    import('../../../scripts/scripts.js');
+
+    let b_productswithselectors;
+
+    beforeEach(() => {
+      b_productswithselectors = document.querySelector('.b-productswithselectors-container');
+    });
+
+    test('check if products have been added to the window.productsListCount', () => {
+        expect(window.productsListCount).toBeTruthy();
+    });
+
+    test('check if the number of servers and mailboxes changes based on number of devices selected', () => {
+      const UsersSelectorBox = b_productswithselectors.querySelectorAll('.selectorBox')[0];
+    
+      // Set the target value on the selector
+      UsersSelectorBox.value = '11';
+      // Dispatch the change event
+      UsersSelectorBox.dispatchEvent(new Event('change'));
+
+      const triggerValue = UsersSelectorBox.value;
+      const fileServers1stProd = Math.ceil((Number(triggerValue)) * 0.3);
```

Comment:
this is hardcoded logic in the test, that means this test will never fail

Reply:
hardcoded values